cmd/tailscale/cli/up.go

```diff
@@ -278,7 +278,9 @@ func prefsFromUpArgs(upArgs upArgsT, warnf logger.Logf, st *ipnstate.Status, goo
 		prefs.NetfilterMode = preftype.NetfilterOff
 	}
 	if upArgs.exitNodeIP != "" {
-		if err := prefs.SetExitNodeIP(upArgs.exitNodeIP, st); err != nil {
+		if expr, useAutoExitNode := ipn.ParseAutoExitNodeStr(upArgs.exitNodeIP); useAutoExitNode {
+			prefs.AutoExitNode = expr
```

Comment:
:thinking: Just to double-check, I don’t understand how `ipn.AnyExitNode` interacts with the system. Like, what happens what you pass in `tailscale set --exit-node=any:foo`?

Reply:
Oops, yes, I meant `auto:foo`. Thanks for clarifying this, I had noticed this behaviour in https://github.com/tailscale/tailscale/pull/16458#discussion_r2183497436, but I had lost context in the interim.